samcli/lib/telemetry/project_metadata.py

```diff
@@ -0,0 +1,108 @@
+"""
+Creates and encrypts metadata regarding SAM CLI projects.
+"""
+
+from os import getcwd
+from os.path import basename
+import re
+import subprocess
+from uuid import uuid5, NAMESPACE_URL
+
+from samcli.cli.global_config import GlobalConfig
+
+
+def get_git_remote_origin_url():
+    """
+    Retrieve an encrypted version of the project's git remote origin url, if it exists.
+
+    Returns
+    -------
+    str | None
+        A UUID5 encrypted string of the git remote origin url, formatted such that the
+        encrypted value follows the pattern <hostname>/<owner>/<project_name>.git.
+        If telemetry is opted out of by the user, or the `.git` folder is not found
+        (the directory is not a git repository), returns None
+    """
+    if not bool(GlobalConfig().telemetry_enabled):
+        return None
+
+    git_url = None
+    try:
+        runcmd = subprocess.run(
+            ["git", "config", "--get", "remote.origin.url"], capture_output=True, shell=True, check=True, text=True
+        )
+        metadata = _parse_remote_origin_url(str(runcmd.stdout))
+        git_url = "/".join(metadata) + ".git"  # Format to <hostname>/<owner>/<project_name>.git
+    except subprocess.CalledProcessError:
+        return None  # Not a git repo
+
+    return _encrypt_value(git_url)
+
+
+def get_project_name():
+    """
+    Retrieve an encrypted version of the project's name, as defined by the .git folder (or directory name if no .git).
+
+    Returns
+    -------
+    str | None
+        A UUID5 encrypted string of either the name of the project, or the name of the
+        current directory that the command is running in.
+        If telemetry is opted out of by the user, returns None
+    """
+    if not bool(GlobalConfig().telemetry_enabled):
+        return None
+
+    project_name = ""
+    try:
+        runcmd = subprocess.run(
+            ["git", "config", "--get", "remote.origin.url"], capture_output=True, shell=True, check=True, text=True
```

Comment:
Does this still work if the remote's name isn't "origin"?

Reply:
Unfortunately it would not, and it would return `None` if there is no "origin". However, it would be safe to assume the default remote name is "origin", since that's often the convention used in Git projects.